modular_zubbers/code/modules/status_indicators/status_indicator.dm

```diff
@@ -54,37 +56,16 @@ GLOBAL_LIST_INIT(potential_indicators, list(
 
 /datum/component/status_indicator/RegisterWithParent()
 	attached_mob = parent
-	// The Basics
-	RegisterSignal(parent, COMSIG_LIVING_DEATH, PROC_REF(cut_indicators_overlays))
-	RegisterSignal(parent, COMSIG_LIVING_LIFE, PROC_REF(status_indicator_evaluate))
-	// When things actually happen
-	RegisterSignal(parent, COMSIG_LIVING_STATUS_STUN, PROC_REF(status_indicator_evaluate))
-	RegisterSignal(parent, COMSIG_LIVING_STATUS_KNOCKDOWN, PROC_REF(status_indicator_evaluate))
-	RegisterSignal(parent, COMSIG_LIVING_STATUS_PARALYZE, PROC_REF(status_indicator_evaluate))
-	RegisterSignal(parent, COMSIG_LIVING_STATUS_IMMOBILIZE, PROC_REF(status_indicator_evaluate))
-	RegisterSignal(parent, COMSIG_LIVING_STATUS_UNCONSCIOUS, PROC_REF(status_indicator_evaluate))
-	RegisterSignal(parent, COMSIG_MOB_LOGIN, PROC_REF(apply_pref_on_login))
-
-/datum/component/status_indicator/proc/apply_pref_on_login()
-	var/atom/movable/screen/plane_master/rendering_plate/game_plate/local_status = locate() in attached_mob.client.screen
-	if(local_status)
-		. = attached_mob.client.prefs.read_preference(/datum/preference/toggle/enable_status_indicators)
-		local_status.alpha = (.) ? 255 : 0
+	RegisterSignals(parent, list(COMSIG_LIVING_LIFE, COMSIG_LIVING_STATUS_STUN, COMSIG_LIVING_STATUS_KNOCKDOWN, COMSIG_LIVING_STATUS_PARALYZE, COMSIG_LIVING_STATUS_IMMOBILIZE, COMSIG_LIVING_STATUS_UNCONSCIOUS), PROC_REF(status_indicator_evaluate))
+
+
+/datum/component/status_indicator/Destroy()
+	. = ..()
```

Comment:
Call parent at the end of your Destroy override not before, you do special handling for this type before you do the rest of the handling